### PR TITLE
Added dijkstra_algorithm.js

### PR DESCRIPTION
**Pull request by LaxmiNyamagoud**
**Hi!**  
Thanks for considering for contributing to this ever-growing list of algorithm and data structure implementations.

#534

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    var visited = [];
+
+    //While there are nodes left to visit
+    while (true) {
+        //find the node with the currently shortest distance from the start node
+        var shortestDistance = Number.MAX_VALUE;
+        var shortestIndex = -1;
+        for (var i = 0; i < graph.length; i++) {
+            if (distance[i] < shortestDistance && !visited[i]) {
+                shortestDistance = distance[i];
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:23]: Codacy found an issue: ['i' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    //it contains whether a node was already visited
+    var visited = [];
+
+    //While there are nodes left to visit
+    while (true) {
+        //find the node with the currently shortest distance from the start node
+        var shortestDistance = Number.MAX_VALUE;
+        var shortestIndex = -1;
+        for (var i = 0; i < graph.length; i++) {
+            if (distance[i] < shortestDistance && !visited[i]) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:22]: Codacy found an issue: ['i' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            if (graph[shortestIndex][i] !== 0 && distance[i] > distance[shortestIndex] + graph[shortestIndex][i]) {
+                //...Save this path as new shortest path.
+                distance[i] = distance[shortestIndex] + graph[shortestIndex][i];
+                console.log("Updating distance of node " + i + " to " + distance[i]);
+            }
+        }
+        // Lastly we finished with this node.
+        visited[shortestIndex] = true;
+        console.log("Visited nodes: " + visited);
+        console.log("Currently lowest distance: " + distance);
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:47]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    //Initial distance is "Infinity"
+    for (var i = 0; i < graph.length; i++) distance[i] = Number.MAX_VALUE;
+    //The distance from the start node to itself is 0
+    distance[start] = 0;
+
+    //it contains whether a node was already visited
+    var visited = [];
+
+    //While there are nodes left to visit
+    while (true) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:17]: Codacy found an issue: [Unexpected constant condition.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            return distance;
+        }
+
+        //for all neighboring nodes....
+        for (var i = 0; i < graph[shortestIndex].length; i++) {
+            //...if the path over this edge is shorter...
+            if (graph[shortestIndex][i] !== 0 && distance[i] > distance[shortestIndex] + graph[shortestIndex][i]) {
+                //...Save this path as new shortest path.
+                distance[i] = distance[shortestIndex] + graph[shortestIndex][i];
+                console.log("Updating distance of node " + i + " to " + distance[i]);
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:41]: Codacy found an issue: ['i' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+        console.log("Visiting node " + shortestDistance + " with current distance " + shortestDistance);
+
+        if (shortestIndex === -1) {
+            // There was no node not yet visited
+            return distance;
+        }
+
+        //for all neighboring nodes....
+        for (var i = 0; i < graph[shortestIndex].length; i++) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:36]: Codacy found an issue: ['i' is already defined.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
+/*
+* Dijkstra's single source shortest path algorithm in JavaScript.
+ */
+ const dijkstraAlgorithm = function (graph, start) {
+
+    //the distances from the start node to all other nodes
+    var distance = [];
+    //Initial distance is "Infinity"
+    for (var i = 0; i < graph.length; i++) distance[i] = Number.MAX_VALUE;
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:9]: Codacy found an issue: ['i' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+    //it contains whether a node was already visited
+    var visited = [];
+
+    //While there are nodes left to visit
+    while (true) {
+        //find the node with the currently shortest distance from the start node
+        var shortestDistance = Number.MAX_VALUE;
+        var shortestIndex = -1;
+        for (var i = 0; i < graph.length; i++) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:21]: Codacy found an issue: ['i' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js (hunk 7 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:9]: Codacy found an issue: [Expected { after 'for' condition.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+            // There was no node not yet visited
+            return distance;
+        }
+
+        //for all neighboring nodes....
+        for (var i = 0; i < graph[shortestIndex].length; i++) {
+            //...if the path over this edge is shorter...
+            if (graph[shortestIndex][i] !== 0 && distance[i] > distance[shortestIndex] + graph[shortestIndex][i]) {
+                //...Save this path as new shortest path.
+                distance[i] = distance[shortestIndex] + graph[shortestIndex][i];
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:40]: Codacy found an issue: ['i' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+    //While there are nodes left to visit
+    while (true) {
+        //find the node with the currently shortest distance from the start node
+        var shortestDistance = Number.MAX_VALUE;
+        var shortestIndex = -1;
+        for (var i = 0; i < graph.length; i++) {
+            if (distance[i] < shortestDistance && !visited[i]) {
+                shortestDistance = distance[i];
+                shortestIndex = i;
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:24]: Codacy found an issue: ['i' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
+/*
+* Dijkstra's single source shortest path algorithm in JavaScript.
+ */
+ const dijkstraAlgorithm = function (graph, start) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:4]: Codacy found an issue: [Function has a complexity of 10. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js (hunk 8 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:21]: Codacy found an issue: ['i' is already defined.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+            //...if the path over this edge is shorter...
+            if (graph[shortestIndex][i] !== 0 && distance[i] > distance[shortestIndex] + graph[shortestIndex][i]) {
+                //...Save this path as new shortest path.
+                distance[i] = distance[shortestIndex] + graph[shortestIndex][i];
+                console.log("Updating distance of node " + i + " to " + distance[i]);
+            }
+        }
+        // Lastly we finished with this node.
+        visited[shortestIndex] = true;
+        console.log("Visited nodes: " + visited);
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:46]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+        var shortestDistance = Number.MAX_VALUE;
+        var shortestIndex = -1;
+        for (var i = 0; i < graph.length; i++) {
+            if (distance[i] < shortestDistance && !visited[i]) {
+                shortestDistance = distance[i];
+                shortestIndex = i;
+            }
+        }
+
+        console.log("Visiting node " + shortestDistance + " with current distance " + shortestDistance);
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:28]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js (hunk 5 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:41]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+        if (shortestIndex === -1) {
+            // There was no node not yet visited
+            return distance;
+        }
+
+        //for all neighboring nodes....
+        for (var i = 0; i < graph[shortestIndex].length; i++) {
+            //...if the path over this edge is shorter...
+            if (graph[shortestIndex][i] !== 0 && distance[i] > distance[shortestIndex] + graph[shortestIndex][i]) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:38]: Codacy found an issue: ['i' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js (hunk 6 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:36]: Codacy found an issue: ['i' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+  
+    const result = {
+      distance: costs.end,
+      path: optimalPath
+    };
+  
+    return result;
+  };
+  
+  console.log(shortestPath(question));
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:74]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+        if (costs[n] > newCost) {
+          costs[n] = newCost;
+          parents[n] = node;
+        }
+      }
+      processed.push(node);
+      node = shortestCostNode(costs, processed);
+    }
+  
+    let optimalPath = ['end'];
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:58]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+  
+    // track nodes that have already been processed
+    const processed = [];
+  
+    let node = shortestCostNode(costs, processed);
+  
+    while (node) {
+      let cost = costs[node];
+      let children = graph[node];
+      for (let n in children) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:43]: Codacy found an issue: [The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+        if (!processed.includes(node)) {
+          lowest = node;
+        }
+      }
+      return lowest;
+    }, null);
+  };
+  
+  // this function returns the minimum cost and path to reach end 
+  const shortestPath = (graph) => {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:25]: Codacy found an issue: [Arrow function has a complexity of 7. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+  
+    // track lowest cost to reach each node
+    const costs = Object.assign({end: Infinity}, graph.start);
+  
+    const parents = {end: null};
+    for (let child in graph.start) {
+      parents[child] = 'start';
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:32]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+  };
+  
+  // this function returns the minimum cost and path to reach end 
+  const shortestPath = (graph) => {
+  
+    // track lowest cost to reach each node
+    const costs = Object.assign({end: Infinity}, graph.start);
+  
+    const parents = {end: null};
+    for (let child in graph.start) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:31]: Codacy found an issue: [The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

**Comment by RavinduSachintha** [comment]
> I have made changes in the program

@LaxmiNyamagoud still there are some required changes for fixing the mentioned issues. Please work on them. :+1:

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js (hunk 16 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:58]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js (hunk 18 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:25]: Codacy found an issue: [Arrow function has a complexity of 7. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js (hunk 19 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:32]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js (hunk 17 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:43]: Codacy found an issue: [The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js (hunk 15 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:74]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js (hunk 20 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:31]: Codacy found an issue: [The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

**Comment by LaxmiNyamagoud** [comment]
> > I have made changes in the program
> 
> @LaxmiNyamagoud still there are some required changes for fixing the mentioned issues. Please work on them. 👍

ok

**Comment by LaxmiNyamagoud** [comment]
I have made changes just check it

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+  function shortestCostNode (costs, processed)  {
+    return Object.keys(costs).reduce((lowest, node) => {
+      if (lowest === null || costs[node] < costs[lowest]) {
+        if (!processed.includes(node)) {
+          lowest = node;
+        }
+      }
+      return lowest;
+    }, null);
+  };
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:22]: Codacy found an issue: [Unnecessary semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+
+  // this function returns the minimum cost and path to reach end
+  function shortestPath(graph) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:25]: Codacy found an issue: [Function 'shortestPath' has a complexity of 9. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

**Comment by LaxmiNyamagoud** [comment]
once again check it

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 23]
```diff
@@ -0,0 +1,76 @@
...
+        if (!processed.includes(node)) {
+          lowest = node;
+        }
+      }
+      return lowest;
+    }, null);
+  }
+
+  // this function returns the minimum cost and path to reach end
+  function shortestPath(graph) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:25]: Codacy found an issue: [Function 'shortestPath' has a complexity of 9. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 24]
```diff
@@ -0,0 +1,71 @@
...
+          lowest = node;
+        }
+      }
+      return lowest;
+    }, null);
+  }
+
+  // this function returns the minimum cost and path to reach end
+  /*eslint complexity: ["error", 7]*/
+  function shortestPath(graph) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:26]: Codacy found an issue: [Function 'shortestPath' has a complexity of 9. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+        if (!processed.includes(node)) {
+          lowest = node;
+        }
+      }
+      return lowest;
+    }, null);
+  }
+
+  // this function returns the minimum cost and path to reach end
+  function shortestPath(graph) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:25]: Codacy found an issue: [Function 'shortestPath' has a complexity of 9. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:25]: Codacy found an issue: [Function 'shortestPath' has a complexity of 9. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

Comment (Danushka96) [reply algorithms/gr-dij/javascript/dijkstra_algorithm.js:25]: Hi @LaxmiNyamagoud
Please simplify the function by dividing it into different small functions
Thanks

Comment (LaxmiNyamagoud) [reply algorithms/gr-dij/javascript/dijkstra_algorithm.js:25]: Ok

Comment (LaxmiNyamagoud) [reply algorithms/gr-dij/javascript/dijkstra_algorithm.js:25]: ok,Thanks!

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 26]
```diff
@@ -0,0 +1,67 @@
...
+
+    // track lowest cost to reach each node
+    const costs = Object.assign({end: Infinity}, graph.start);
+    const parents = {end: null};
+    for (let child in graph.start) {
+      if((graph.start).hasOwnProperty(child)){
+        parents[child] = 'start';
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:32]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

File: algorithms/gr-dij/javascript/dijkstra_algorithm.js [hunk 27]
```diff
@@ -0,0 +1,67 @@
...
+        if (!processed.includes(node)) {
+          lowest = node;
+        }
+      }
+      return lowest;
+    }, null);
+  }
+
+  // this function returns the minimum cost and path to reach end
+  function shortestPath(graph) {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/javascript/dijkstra_algorithm.js:25]: Codacy found an issue: [Function 'shortestPath' has a complexity of 9. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)

**Comment by LaxmiNyamagoud** [comment]
when you merge this algoritm?

**Comment by LaxmiNyamagoud** [comment]
when you merge this algorithm? and label as hacktoberfest-accepted

**Review by Danushka96 — APPROVED** [review]

**Comment by Danushka96** [comment]
Hi @LaxmiNyamagoud 
Can you update the main Readme.md algorithm chart with the algorithm you implemented?
Just update the readme file and send the commit to this PR,
Thanks

**Comment by LaxmiNyamagoud** [comment]
> Hi @LaxmiNyamagoud
> Can you update the main Readme.md algorithm chart with the algorithm you implemented?
> Just update the readme file and send the commit to this PR,
> Thanks

ok I will do it
Thanks!

**Comment by LaxmiNyamagoud** [comment]
I have updated Readme.md file

**Comment by Danushka96** [comment]
@all-contributors please add @LaxmiNyamagoud for code and doc

**Comment by Danushka96** [comment]
Hi @LaxmiNyamagoud 
Thanks for your contribution!

**Comment by LaxmiNyamagoud** [comment]
> Hi @LaxmiNyamagoud
> Thanks for your contribution!

Thanks!

**Pull request merged**